hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -828,22 +832,29 @@ private CopyObjectResponse copyObject(String copyHeader,
     }
     int pos = header.indexOf('/');
     if (pos == -1) {
-      OS3Exception ex = S3ErrorTable.newError(S3ErrorTable
-          .INVALID_ARGUMENT, header);
+      OS3Exception ex = S3ErrorTable.newError(INVALID_ARGUMENT, header);
       ex.setErrorMessage("Copy Source must mention the source bucket and " +
           "key: sourcebucket/sourcekey");
       throw ex;
     }
 
-    return Pair.of(header.substring(0, pos), header.substring(pos + 1));
+    try {
+      String bucket = header.substring(0, pos);
```

Comment:
Question:
What is the reason not to do urlDecode on the copyHeader if it is urlEncoded?